openid-4-verifiable-presentations-1_0.md

```diff
@@ -432,54 +432,32 @@ Note: "https://self-issued.me/v2" is a symbolic string and can be used as an `au
+If the Client is not pre-registered and if the Authorization Request is unsigned, the Wallet MUST check if the `client_id` matches the `redirect_uri` or `response_uri` parameter.
 
-* `x509_san_dns`: When the Client Identifier Scheme is `x509_san_dns`, the Client Identifier MUST be a DNS name and match a `dNSName` Subject Alternative Name (SAN) [@!RFC5280] entry in the leaf certificate passed with the request. The request MUST be signed with the private key corresponding to the public key in the leaf X.509 certificate of the certificate chain added to the request in the `x5c` JOSE header [@!RFC7515] of the signed request object. The Wallet MUST validate the signature and the trust chain of the X.509 certificate. All Verifier metadata other than the public key MUST be obtained from the `client_metadata` parameter. If the Wallet can establish trust in the Client Identifier authenticated through the certificate, e.g. because the Client Identifier is contained in a list of trusted Client Identifiers, it may allow the client to freely choose the `redirect_uri` value. If not, the FQDN of the `redirect_uri` value MUST match the Client Identifier.
+It is a Wallet-policy decision on how to authenticate the Request Object, whether to accept unsigned Authorization Requests and whether the linkage between the `redirect_uri` or `response_uri` and the Client is verified.
 
-* `x509_san_uri`: When the Client Identifier Scheme is `x509_san_uri`, the Client Identifier MUST be a URI and match a `uniformResourceIdentifier` Subject Alternative Name (SAN) [@!RFC5280] entry in the leaf certificate passed with the request. The request MUST be signed with the private key corresponding to the public key in the leaf X.509 certificate of the certificate chain added to the request in the `x5c` JOSE header [@!RFC7515] of the signed request object. The Wallet MUST validate the signature and the trust chain of the X.509 certificate. All Verifier metadata other than the public key MUST be obtained from the `client_metadata` parameter. If the Wallet can establish trust in the Client Identifier authenticated through the certificate, e.g. because the Client Identifier is contained in a list of trusted Client Identifiers, it may allow the client to freely choose the `redirect_uri` value. If not, the `redirect_uri` value MUST match the Client Identifier.
+Note: To prevent downgrade attacks, a Client can choose to include certain parameters in the request object or not.
 
-To use `client_id_scheme` values `entity_id`, `did`, `verifier_attestation`, `x509_san_dns`, and `x509_san_uri`, Verifiers MUST be confidential clients. This might require changes to the technical design of native apps as such apps are typically public clients.
+The following is a list of methods defined by this specification that MAY be used by the Wallet to authenticate the Request Object against the `client_id` in the Request Object and verify the `redirect_uri` or `response_uri` belong to the Client:
```

Comment:
@selfissued: does the above MAY express that even if the methods below contain a MUST, they only apply in case the Wallet chose to apply one of these rules?

Reply:
I suggested changing "MAY be used" to "are used".

Reply:
That makes sense